pycloudlib/lxd/cloud.py

```diff
@@ -511,3 +494,220 @@ def _find_image(self, release, arch=LOCAL_UBUNTU_ARCH, daily=True):
+        )
+
+        return [profile_name]
+
+    def _prepare_command(
+            self, name, release, ephemeral=False, network=None, storage=None,
+            inst_type=None, profile_list=None, user_data=None,
+            config_dict=None):
+        """Build a the command to be used to launch the LXD instance.
+
+        Args:
+            name: string, what to call the instance
+            release: string, [<remote>:]<release>, what release to launch
+                     (default remote: )
+            ephemeral: boolean, ephemeral, otherwise persistent
+            network: string, optional, network name to use
+            storage: string, optional, storage name to use
+            inst_type: string, optional, type to use
+            profile_list: list, optional, profile(s) to use
+            user_data: used by cloud-init to run custom scripts/configuration
+            config_dict: dict, optional, configuration values to pass
+
+        Returns:
+            A list of string representing the command to be run to
+            launch the LXD instance.
+        """
+        if not profile_list:
+            profile_list = self._build_necessary_profiles(release=release)
+
+        cmd = super()._prepare_command(
+            name=name,
+            release=release,
+            ephemeral=ephemeral,
+            network=network,
+            storage=storage,
+            inst_type=inst_type,
+            profile_list=profile_list,
+            user_data=user_data,
+            config_dict=config_dict
+        )
+
+        cmd.append("--vm")
+
+        return cmd
+
+    def _get_image_hash_key(self, release=None):
+        """Get the correct hash key to be used to launch LXD instance.
+
+        When query simplestreams for image information, we receive a
+        dictionary of metadata. In that metadata we have the necessary
+        information to allows us to launch the required image. However,
+        we must know which key to use in the metadata dict to allows
+        to launch the image.
+
+        Args:
+            release: string, optional, Ubuntu release
+
+        Returns
+            A string specifying which key of the metadata dictionary
+            should be used to launch the image.
+        """
+        return self.VM_HASH_KEY
+
+    def _search_for_image(
+        self, remote, daily, release, arch=LOCAL_UBUNTU_ARCH
+    ):
+        """Find the LXD fingerprint in a given remote.
+
+        Args:
+            remote: string, remote to prepend to image_id
+            daily: boolean, search on daily remote
+            release: string, Ubuntu release to look for
+            arch: string, architecture to use
+
+        Returns:
+            string, LXD fingerprint of latest image
+
+        """
+        if release == "xenial":
+            # xenial needs to launch images:ubuntu/16.04/cloud
+            # because it contains the HWE kernel which has vhost-vsock support
+            return self.XENIAL_IMAGE_VSOCK_SUPPORT
```

Comment:
This is a different image to the one that consumers will expect (it's from `images:`, instead of from `ubuntu:`); I'd like to see us log something to give people a better chance of grokking what's going on.